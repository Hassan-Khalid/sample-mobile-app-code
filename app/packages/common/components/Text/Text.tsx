import {Text as NText,TextStyle, TextProps} from "react-native"
import React from "react";
import {useTranslation} from 'react-i18next';
import { AnyType } from 'common/declarations/types';
import AppColors from 'common/assets/colors/AppColors';

interface _TextProps extends TextProps{
  style?:TextStyle,
  children:React.ReactNode | AnyType,
}

const Text = (props:_TextProps)=>{
  const {t} = useTranslation();
  return(
    <NText style={{...props.style,...{color:AppColors.BACKGROUND}}} {...props} >
      {t(props.children)}
    </NText>
  )
}

export default Text;