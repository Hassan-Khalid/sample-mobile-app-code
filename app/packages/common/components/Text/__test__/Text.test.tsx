import React from 'react';
import { render } from '@testing-library/react-native';
import Text from '../Text';

describe('Text Component', () => {
  it('renders correctly empty', () => {
    // exercise
    const wrapper = render(<Text/>);
    
    // post-conditions
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  
  it('should render with child', () => {
    // exercise
    const wrapper = render(<Text>'hello'</Text>);
    
    // post-conditions
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
