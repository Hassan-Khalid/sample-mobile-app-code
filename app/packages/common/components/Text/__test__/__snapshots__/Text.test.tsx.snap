// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component renders correctly empty 1`] = `
<Text
  style={
    {
      "color": {
        "dynamic": {
          "dark": "#172B4D",
          "highContrastDark": undefined,
          "highContrastLight": undefined,
          "light": "#172B4D",
        },
      },
    }
  }
/>
`;

exports[`Text Component should render with child 1`] = `
<Text
  style={
    {
      "color": {
        "dynamic": {
          "dark": "#172B4D",
          "highContrastDark": undefined,
          "highContrastLight": undefined,
          "light": "#172B4D",
        },
      },
    }
  }
>
  'hello'
</Text>
`;
